XVim/XVimReplaceEvaluator.m

```diff
@@ -0,0 +1,95 @@
+//
+//  XVimReplaceEvaluator.m
+//  XVim
+//
+//  Created by Martin Conte Mac Donell on 12/14/14.
+//
+
+#import "XVimReplaceEvaluator.h"
+#import "XVimWindow.h"
+
+@interface XVimInsertEvaluator()
+
+@property (nonatomic) NSRange startRange;
+@property (nonatomic) BOOL movementKeyPressed;
+
+- (NSString*)insertedText;
+- (BOOL)windowShouldReceive:(SEL)keySelector;
+
+@end
+
+@implementation XVimReplaceEvaluator {
+    BOOL _oneCharMode;
+}
+
+- (float)insertionPointHeightRatio{
+    return 0.25;
+}
+
+- (float)insertionPointWidthRatio{
+    return 1.0;
+}
+
+- (float)insertionPointAlphaRatio{
+    return 0.5;
+}
+
+- (id)initWithWindow:(XVimWindow*)window oneCharMode:(BOOL)oneCharMode mode:(XVimInsertionPoint)mode{
+    self = [super initWithWindow:window mode:mode];
+    if (self) {
+        _oneCharMode = oneCharMode;
+    }
+    return self;
+}
+
+- (NSString*)modeString{
+	return @"-- REPLACE --";
+}
+
```

Comment:
I think you want to implement `-mode` as well. Which probably means adding a value to the `XVIM_MODE` enum.